Fortnox.NET/WebSockets/FortnoxWebSocketClient.cs

```diff
@@ -0,0 +1,273 @@
+        /// <param name="bufferSize">Optional target buffer size when recieving messages from the socket connection.</param>
+        public FortnoxWebSocketClient(List<string> accessTokens, string clientSecret, int bufferSize = DEFAULT_BUFFER_SIZE)
+        {
+            this._accessTokens = new List<string>(accessTokens);
+
+            this._clientSecret = clientSecret;
+            this._fortnoxWebSocketURI = new Uri(ApiEndpoints.WebSocketURI);
+
+            this._webSocket = new ClientWebSocket();
+
+            this._topics = new List<string>();
+
+            this._bufferSize = bufferSize;
+        }
+
+        /// <summary>
+        /// Sends the specified data on the underlying <see cref="System.Net.WebSockets.ClientWebSocket"/> and then recieves the next message as an asynchronous operation.
+        /// Then deserializes the recieved data as the specified .NET type.
+        /// </summary>
+        /// <typeparam name="T">The .NET type to deserialize the recieved message to.</typeparam>
+        /// <param name="message">The data to send.</param>
+        /// <returns>The deserialized message.</returns>
+        private async Task<T> SendAndRecieveOnce<T>(string message)
+        {
+            var messageBuffer = new ArraySegment<byte>(Encoding.ASCII.GetBytes(message));
+            await _webSocket.SendAsync(messageBuffer, WebSocketMessageType.Text, true, CancellationToken.None);
+            
+            var finished = false;
+            var resultString = "";
+
+            while (!finished)
+            {
+                if (_webSocket.State == WebSocketState.Open)
+                {
+                    var resultBuffer = new byte[this._bufferSize];
+                    var result = await _webSocket.ReceiveAsync(new ArraySegment<byte>(resultBuffer), CancellationToken.None);
+
+                    if (result.MessageType == WebSocketMessageType.Close)
+                    {
+                        await _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty, CancellationToken.None);
+                        throw new WebSocketException($"Websocket connection closed unexpectedly. Reason: {result.CloseStatus}");
+                    }
+                    else
+                    {
+                        resultString += Encoding.ASCII.GetString(resultBuffer);
+                        finished = result.EndOfMessage;
+                    }
+                }
+                else
+                {
+                    throw new WebSocketException("Unable to recieve as Websocket is closed.");
+                }
+            }
+
+            try
+            {
+                var deserializedResult = JsonConvert.DeserializeObject<T>(resultString, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+                return deserializedResult;
+            }
+            catch (Exception e)
+            {
+                throw e;
```

Comment:
since we aren't doing something special here, what's the point of wrapping in a try...catch and re-throwing?